llvm/include/llvm/Testing/Support/SupportHelpers.h

```diff
@@ -172,19 +172,23 @@ class TempDir {
 /// The link may be a soft or a hard link, depending on the platform.
```

Comment:
Given this comment, could we update the implementation to support creating a soft directory link on Windows (probably a directory junction rather than symbolic link), so we don't have to ignore errors?

Reply:
I got the impression that there's a reason not to support soft links on windows in the FileSystem API from the upstream LLVM as symlinks would require a certain system privilege on windows. Based on that, I disabled the test code paths that relies on symlinks for now. I'm not sure if symlinks are necessary for the CAS support or this particular PR (whether that can be future work after this PR). I updated the PR description to clarify this point. @cachemeifyoucan do you have thoughts on this?

Reply:
I think for test coverage, that is fine. We don't need to test soft link on windows for CAS file system.

For CAS File system, the goal is to create an abstract file system representation in CAS that can mimic the real file system. I don't think we need to replicate all aspects of the file system if our tools that is using the CAS file system do not relied on those features. For example, we can easily model symlinks by using two files (since they de-duplicate the CAS anyway) as long as compiler is not trying to check and see if one of them is a symlink or not.

Reply:
Got it 👍